### PR TITLE
Build Docker images with conda-forge base too

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Comes from https://github.com/conda/conda-libmamba-solver/pull/120.

This PR provides an additional Docker image for CI testing. It just provides one more option to debug installation issues. We are often mixing channels (technically unsupported) to provide extra dependencies not available on `defaults` (project not packaged, version out of date, etc).

This would help maintainers choose if they want to also test with a pure conda-forge setup, or maybe just for local debugging when trying to reproduce issues users with a pure conda-forge setup are facing.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: .github/workflows/images.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ on:
   # NOTE: github.event context is push payload:
   # https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push
   push:
+    branches:
+      - main
```
Comment (jezdez) [inline .github/workflows/images.yml:19]: That would take away the ability to update the image in a branch, which we've used a few times, making it easier to handle feature branches without changing the main image.

File: .github/workflows/images.yml [hunk 2]
```diff
@@ -76,14 +83,14 @@ jobs:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
           flavor: |
             latest=false
-            suffix=-linux-python${{ matrix.python }}
+            suffix=-linux-python${{ matrix.python }}${{ matrix.channel == 'conda-forge' && '-conda-forge' || '' }}
```
Comment (jezdez) [inline .github/workflows/images.yml:86]: I don't think we've agreed to maintain the main image name that way, I could just see this being 
```suggestion
            suffix=-linux-python${{ matrix.python }}-${{ matrix.channel }}
```

File: Dockerfile [hunk 3]
```diff
@@ -19,22 +19,39 @@ FROM --platform=$TARGETPLATFORM debian:stable-slim AS buildbase
...
+            MINICONDA_URL="https://github.com/conda-forge/miniforge/releases/${CONDA_VERSION}/download/Miniforge3-Linux-ppc64le.sh"; \
+        else \
+            echo "Not supported source channel & target architecture: ${conda_docker_default_channel} & ${TARGETARCH}"; \
+            exit 1; \
+        fi; \
     else \
-        echo "Not supported target architecture: ${TARGETARCH}"; \
-        exit 1; \
+        echo "conda_docker_default_channel value ${conda_docker_default_channel} not supported"; \
+        exit 1; \
```
Comment (jezdez) [inline Dockerfile]: @dbast probably has opinions on this, but could we try to move some of this into a script so it's not as painful to look at?

**Comment by travishathaway** [comment]
Hi @jaimergp,

Thanks for the pull request. I would advise also adding a little text about this in our "development environment" docs:

- https://docs.conda.io/projects/conda/en/stable/dev-guide/development-environment.html

File: .github/workflows/images.yml (hunk 2 above)
Comment (jezdez) [reply .github/workflows/images.yml:86]: Hmpf, yeah, true. Okay, fair enough.

**Review by dbast — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**